### PR TITLE
Better error message for unsupported Ruff version

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR adds back the message to suggest the user that the current Ruff binary doesn't support the native server.

This was removed from the Python script in #511.

fixes: #476 

## Test Plan

Install an older Ruff version to test this out using `pipx install --force "ruff==0.3.3"`.

Screenshot 2024-07-03 at 14 50 41

And, we've another log message which displays the Ruff version similar to `ruff-lsp`:
```
2024-07-03 16:00:03.089 [info] Using the Ruff binary: /Users/dhruv/work/astral/ruff-lsp/.venv/bin/ruff
2024-07-03 16:00:03.094 [info] Found Ruff 0.4.10 at /Users/dhruv/work/astral/ruff-lsp/.venv/bin/ruff
2024-07-03 16:00:03.095 [info] Server run command: /Users/dhruv/work/astral/ruff-lsp/.venv/bin/ruff server --preview
```

### CODE REVIEW

File: src/common/server.ts [hunk 1]
```diff
@@ -48,6 +49,54 @@ function executeCommand(command: string): Promise<string> {
...
+ * Convert a version object to a string.
+ */
+function versionToString(version: VersionInfo): string {
+  return `${version.major}.${version.minor}.${version.patch}`;
+}
+
+/**
+ * The minimum version of the Ruff executable that supports the native server.
+ */
+const MINIMUM_RUFF_SERVER_VERSION: VersionInfo = { major: 0, minor: 3, patch: 5 };
```
Comment (dhruvmanila) [inline src/common/server.ts:68]: This comes from https://github.com/astral-sh/ruff-vscode/pull/443 although the minimum supported version if actually `0.3.3` but I guess this is the version we moved away from using the bundled version.

Comment (T-256) [reply src/common/server.ts:68]: As explained in https://github.com/astral-sh/ruff-vscode/issues/508 there are also commands/features that unsupported by old version of nativeServer and/or ruff-lsp. should we declare version constant for them? e.g. `ruff.printDebugInformation` which was added in https://github.com/astral-sh/ruff/pull/11831 for nativeServer

Comment (dhruvmanila) [reply src/common/server.ts:68]: Good point. I intend to implement the warning part for the settings in https://github.com/astral-sh/ruff-vscode/pull/505.

> `ruff.printDebugInformation` which was added in [astral-sh/ruff#11831](https://github.com/astral-sh/ruff/pull/11831) for nativeServer

Regarding this command, I think we should only register a VS Code command only if it's available in the server capabilities (https://github.com/astral-sh/ruff/pull/11850). This way we don't need to check the Ruff version for at least the server capabilities.

Comment (dhruvmanila) [reply src/common/server.ts:68]: I've opened #513 to keep track of it.

**Review by MichaReiser — APPROVED** [review]

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**